### PR TITLE
Update product order

### PR DESCRIPTION
**Pull request by gsim1903**
Create test spec to update an order; 
Def update method in orders_controller
Update order model to def serialised json rsponse
Modify order controller to return json response for multiple order items added
Test all the above

### CODE REVIEW

**Comment by gsim1903** [comment]
(https://www.pivotaltracker.com/story/show/182060662)

File: spec/requests/api/orders/update_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    put "/api/orders/#{order.id}", params: { product_id: product_2.id }
+
+    @order = Order.last
+  end
+
+  it { is_expected.to have_http_status :ok }
+
+  it 'response with an order confirmation' do
+    response_body = JSON.parse(response.body)
+    expect(response_body['message']).to eq 'Football added'
```
Comment (maxarvid) [inline spec/requests/api/orders/update_spec.rb]: `response_json` does the same thing with less code

File: spec/requests/api/orders/update_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  let(:product_3) { create(:product, name: 'Scarf') }
+  let(:order) { create(:order) }
+  let!(:order_item_1) { create(:order_item, order:, product: product_2) }
+  let!(:order_item_2) { create(:order_item, order:, product: product_3) }
+
+  subject { response }
+  
+  describe 'succesful order' do
+  before do
+    put "/api/orders/#{order.id}", params: { product_id: product_2.id }
```
Comment (maxarvid) [inline spec/requests/api/orders/update_spec.rb]: Format your code.

File: spec/requests/api/orders/update_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  it 'is expected to contain 3 items' do
+    expect(@order.items.count).to eq 3
+  end
+
+end
+
+  describe 'unsuccesful order' do
+    before do
+      put "/api/orders/#{order.id}", params: { product_id: 999_999 }
```
Comment (maxarvid) [inline spec/requests/api/orders/update_spec.rb]: `999_999` makes me nervous. Consider passing in a string as follows:`'999999'`

File: spec/requests/api/orders/update_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  it 'response with an order confirmation' do
+    response_body = JSON.parse(response.body)
+    expect(response_body['message']).to eq 'Football added'
+  end
+
+  it 'is expected to contain 3 items' do
+    expect(@order.items.count).to eq 3
+  end
+
```
Comment (maxarvid) [inline spec/requests/api/orders/update_spec.rb:33]: Consider adding a test to make sure that the order includes the items you expect it to include. If I read the above correctly, Tshirt is never added.

File: spec/requests/api/orders/update_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
+RSpec.describe 'Update an order with a PUT /api/orders/:id', type: :request do
+
+  
+  let(:user) { create(:user) }
+  let(:product_1) { create(:product, name: 'Tshirt') }
+  let(:product_2) { create(:product, name: 'Football') }
+  let(:product_3) { create(:product, name: 'Scarf') }
+  let(:order) { create(:order) }
+  let!(:order_item_1) { create(:order_item, order:, product: product_2) }
+  let!(:order_item_2) { create(:order_item, order:, product: product_3) }
```
Comment (maxarvid) [inline spec/requests/api/orders/update_spec.rb:8]: Why is there a `:` in the second argument of `create`?

File: spec/models/order_spec.rb [hunk 6]
```diff
@@ -12,6 +12,23 @@
...
+
+  describe 'Instances methods' do
+    describe 'serialized' do
+      let(:product_1) { create(:product, name: 'Tshirt') }
+      let(:product_2) { create(:product, name: 'Football') }
+      let(:product_3) { create(:product, name: 'Scarf') }
+      let(:order) { create(:order) }
+      let!(:order_item_1) { create(:order_item, order:, product: product_1) }
+      let!(:order_item_2) { create(:order_item, order:, product: product_2) }
+      let!(:order_item_3) { create(:order_item, order:, product: product_3) }
```
Comment (maxarvid) [inline spec/models/order_spec.rb:26]: Why is there a `:` in the 2nd argument of `create`?

File: app/controllers/api/orders_controller.rb [hunk 7]
```diff
@@ -3,7 +3,16 @@ def create
...
     order.items.create(product:)
-    render json: { order: }, status: :created
+    render json: { order: order.serialized }, status: :created
+  end
+
+  def update
+    product = Product.find(params[:product_id])
+    order = Order.find(params[:id])
+    order.items.create(product:)
+    render json: { order: order.serialized, message: "#{product.name} added" }, status: 200
```
Comment (maxarvid) [inline app/controllers/api/orders_controller.rb:13]: Format your code.

File: spec/requests/api/orders/create_spec.rb [hunk 8]
```diff
@@ -29,4 +29,11 @@
   it 'is expected to assocaite the product with an order ' do
     expect(@order.products.first).to eq product
   end
+  
+  it 'is expected to include an Order ID in the response ' do
+    response_body = JSON.parse(response.body)
+    expect(response_body['order']['id']).to eq @order.id
```
Comment (maxarvid) [inline spec/requests/api/orders/create_spec.rb]: See comment above on `response_json`

**Review by maxarvid — CHANGES_REQUESTED** [review]
Right on, made some comments. Also, add a link to the PT story in the PR description.

File: spec/requests/api/orders/update_spec.rb (hunk 1 above)
Comment (gsim1903) [reply spec/requests/api/orders/update_spec.rb]: Screenshot 2022-05-04 at 18 45 58

File: spec/requests/api/orders/update_spec.rb (hunk 2 above)
Comment (gsim1903) [reply spec/requests/api/orders/update_spec.rb]: Done

File: spec/requests/api/orders/update_spec.rb (hunk 3 above)
Comment (gsim1903) [reply spec/requests/api/orders/update_spec.rb]: Screenshot 2022-05-04 at 18 46 43

File: app/controllers/api/orders_controller.rb (hunk 7 above)
Comment (gsim1903) [reply app/controllers/api/orders_controller.rb:13]: Updated

File: spec/requests/api/orders/update_spec.rb (hunk 5 above)
Comment (gsim1903) [reply spec/requests/api/orders/update_spec.rb:8]: I understand this creates the order item  for product 2 product 3. @maxarvid can you advise help / advise how to re-write. When I try remove tests all fail

File: spec/requests/api/orders/create_spec.rb (hunk 8 above)
Comment (gsim1903) [reply spec/requests/api/orders/create_spec.rb]: Updated

File: spec/requests/api/orders/update_spec.rb (hunk 4 above)
Comment (gsim1903) [reply spec/requests/api/orders/update_spec.rb:33]: Screenshot 2022-05-04 at 19 33 15

Comment (gsim1903) [reply spec/requests/api/orders/update_spec.rb:33]: Updated, did scarf by accident!

File: spec/requests/api/orders/update_spec.rb (hunk 5 above)
Comment (maxarvid) [reply spec/requests/api/orders/update_spec.rb:8]: So this is not a problem, I was looking for an explanation of why we can set a hash value to nothing. I've made a post in Slack explaining why we can omit the value here.

**Review by maxarvid — APPROVED** [review]
LGTM

**Pull request merged**